nixos/modules/services/misc/snapper.nix

```
@@ -353,4 +353,6 @@ in
       ) (attrNames cfg.configs);
     }
   );
+
+  meta.maintainers = [ maintainers.Djabx ];
```

```suggestion
  meta.maintainers = with lib.maintainers; [ Djabx ];
```
The `with lib;` at top of NixOS module is bad practice considering it leads to `nested with` which can cause issues in static analysis, and lead to ambiguity and some edge cases. In general, it is good practice to avoid it.

I suggest you to not rely on `with lib;` (everywhere), because it should be eventually removed.

---

> The `with lib;` at top of NixOS module is bad practice considering it leads to `nested with` which can cause issues in static analysis, and lead to ambiguity and some edge cases. In general, it is good practice to avoid it.

I'm not really sure to understand the point here, but I believe you :)

Is there any good resources / tutorial on how to write nix files ? And maybe some trap to avoid ?

---

https://nix.dev/guides/best-practices.html#with-scopes

---

To clarify, you can use `with`, just avoid nesting it.